browser/extensions/api/brave_rewards_api.h

```diff
@@ -54,6 +54,22 @@ class BraveRewardsTipTwitterUserFunction
       std::unique_ptr<brave_rewards::ContentSite> publisher_info);
 };
 
+class BraveRewardsTipRedditUserFunction : public UIThreadExtensionFunction {
+ public:
+  BraveRewardsTipRedditUserFunction();
+  DECLARE_EXTENSION_FUNCTION("braveRewards.tipRedditUser", UNKNOWN)
```

Comment:
not specific to this PR, but is `chrome.braverewards` only accessible to the rewards extension? it would be bad if it were generally available to extensions.

Reply:
My understanding is that our code would be sandboxed/isolated from web pages and other extensions with communication only occurring through message API.

Reply:
@jasonrsadler right, but is the rewards extension the only one that can call `chrome.braveRewards`?

Reply:
Currently, it appears to be the only one whitelisted for that namespace:
https://github.com/brave/brave-core/blob/master/common/extensions/api/_api_features.json#L85